app/scripts/directives/blockRegistration.js

```diff
@@ -24,6 +24,36 @@ angular.module('confRegistrationWebApp')
           if ($scope.block.endDateBlockId !== null) {
             DateRangeService.subscribe($scope.block.endDateBlockId, endDateChangeCallback);
           }
+
+          $scope.$watch('answer', function (answer, oldAnswer) {
+            if (angular.isUndefined(answer) || angular.isUndefined(oldAnswer) || angular.equals(answer, oldAnswer)) {
+              return;
+            }
+
+            $rootScope.$broadcast('answerChanged');
```

Comment:
It's because I'm listening to changes on a different instance of this component. What happens here is that whenever an answer is modified, then I need to check if the checkbox, select and radio answers are still valid. So the checkbox/select/radio block component should be notified about changes on other answers.

Reply:
Any way around that? I'm just not sure what you are doing. Why do checkbox, select, and radio components all need to interact with each other?

Reply:
Some Questions have rules that hide/show some of its choices. For example: Depending on what I answer for Question A, then the available choices in the Question B will change. I need to remove these invalid choices from `$scope.answer` on Question B, not on Question A. But the `watch` event happens on Question A. That's why I'm using broadcast.

I didn't find a workaround, because the Question B component has no access to Question A's `$scope.answer` (and I need that in order to know what choices are invalid). The only way I found was to share scopes, but this seems worse.

Reply:
Oh that makes more sense. Didn't know Answer Rules were a thing now... Gosh this sounds complicated.

My concern is that firing `answerChanged` could fire listeners on a whole lot of questions and do it repeatedly if a user is typing. That sounds expensive.

Maybe in the parent component you could have a map with keys of rule ids and values of true/false indicating whether the condition for the rule has been met.  You could make the question that controls the rule condition go update that whether that rule is active or not when its answer changes. Then you could go make the dependent question lookup just that rule id to see if the rule is active and hide stuff accordingly. I guess it would have to go read from the map on every render or have a watcher or something anyways to make it work. But looking up one boolean in a map sounds less expensive than running through all the block answers to see if the rule has been met for each answer (maybe it is implemented better than that, idk). And only updating the map when a rule changes active state would fire changes less than on every answer change since it would effectively filter out irrelevant answer changes. I haven't quite thought this through with all the and/or conditions. Idk if this is feasible. It may be pretty hard to refactor how it's currently implemented.

Idk. What are your thoughts? Any other ideas? Maybe my other comment would be enough and prevent this from happening on every question. Maybe this should be debounced so it's not firing multiple times in quick succession when changes are made. Maybe you could make a question look to see if it has any dependent rules and never broadcast the event if it determines no rules depend on it.

I suppose we could just live with this if there aren't any performance issues. We need to make sure to test on a conference with lots of questions and rules. Just seems like we're making all these components even more interdependent. AngularJS sure doesn't make this easy.

Reply:
Ok, I'll try to refactor it to make it less expensive. Your suggestions should be enough to improve the performance.